internal/provider/provider.go

```diff
@@ -79,8 +86,8 @@ func configure(version string, p *schema.Provider) func(context.Context, *schema
 		if username == "" || password == "" {
 			diags = append(diags, diag.Diagnostic{
 				Severity: diag.Error,
-				Summary:  "Username and password must be set",
-				Detail:   "Currently the provider can only authenticate using username and password based authentication, if both are empty the provider will panic",
+				Summary:  authErrSummary,
+				Detail:   authErrDetail,
```

Comment:
In this case, it's better to have the summary and details in-line rather than pulled out in a `const`.